x-pack/plugins/observability_solution/observability/server/services/slo/transform_generators/synthetics_availability.ts

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { TransformPutTransformRequest } from '@elastic/elasticsearch/lib/api/types';
+import { estypes } from '@elastic/elasticsearch';
+import {
+  ALL_VALUE,
+  syntheticsAvailabilityIndicatorSchema,
+  timeslicesBudgetingMethodSchema,
+} from '@kbn/slo-schema';
+import { getElasticsearchQueryOrThrow, TransformGenerator } from '.';
+import {
+  getSLOTransformId,
+  SLO_DESTINATION_INDEX_NAME,
+  SLO_INGEST_PIPELINE_NAME,
+} from '../../../../common/slo/constants';
+import { getSLOTransformTemplate } from '../../../assets/transform_templates/slo_transform_template';
+import { SyntheticsAvailabilityIndicator, SLO } from '../../../domain/models';
+import { InvalidTransformError } from '../../../errors';
+export class SyntheticsAvailabilityTransformGenerator extends TransformGenerator {
+  public getTransformParams(slo: SLO, spaceId: string): TransformPutTransformRequest {
+    if (!syntheticsAvailabilityIndicatorSchema.is(slo.indicator)) {
+      throw new InvalidTransformError(`Cannot handle SLO of indicator type: ${slo.indicator.type}`);
+    }
+
+    return getSLOTransformTemplate(
+      this.buildTransformId(slo),
+      this.buildDescription(slo),
+      this.buildSource(slo, slo.indicator, spaceId),
+      this.buildDestination(),
+      this.buildGroupBy(slo, slo.indicator),
+      this.buildAggregations(slo),
+      this.buildSettings(slo)
```

Comment:
By defualt the settings will use `@timestamp` for syncing the transform with the raw data. If the synthetics index have `event.ingested` I would suggest to use that instead.